src/test/java/org/opensearch/ad/rest/AnomalyDetectorRestApiIT.java

```diff
@@ -484,6 +487,22 @@ public void testUpdateAnomalyDetectorFlattenResultIndexField() throws Exception
             detector.getLastBreakingUIChangeTime()
         );
 
+        if (isResourceSharingFeatureEnabled()) {
+            Awaitility.await().atMost(Duration.ofSeconds(30)).pollInterval(Duration.ofMillis(200)).until(() -> {
```

Comment:
this is required as adding resource-sharing is handled by index listener in security plugin and requires some time to index resource-sharing record.

Reply:
Behind the scenes, security plugin calls an update operation to create/update the resource sharing record. Since tests run much faster consecutively then real world use-case, we force the tests to wait until resource-sharing record is updated.

It is used in all places after create/update AD or forecaster config is successful.